pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -260,6 +248,22 @@ func (e *aksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 	}
 }
 
+// checkCRDReady checks whether necessary CRD, AKSConfig, has been created yet
+func (e *aksOperatorController) checkCrdReady(cluster *mgmtv3.Cluster) (*mgmtv3.Cluster, error) {
```

Comment:
This method can be moved to `pkg/controllers/management/clusteroperator/utils.go` and changed to take a string parameter as well: `checkCrdReady(cluster *mgmtv3.Cluster, provider string)`.

Then each handler could pass a string, like: `checkCrdReady(cluster, "aks")`.

We try to keep the common code in `pkg/controllers/management/clusteroperator/utils.go`.

Reply:
Sounds good, let me change that.

Reply:
:+1: Changed to common function.